app/controllers/requests_controller.rb

```diff
@@ -0,0 +1,33 @@
+class RequestsController < ApplicationController
+  def create
+    @request = current_user.requests.build(request_params)
+    @books = Book.where(id: session[:book_ids])
+    @books.each do |book|
+      @request.request_items.build(book_id: book.id, quantity: 1)
+    end
+    check_create_save @request
+  end
```

Comment:
```suggestion
  def create
    @request = current_user.requests.build(request_params)
    @books = Book.where(id: session[:book_ids])
    @books.each do |book|
      @request.request_items.build(book_id: book.id, quantity: 1)
    end
    @request.save
  end
```
ri luôn em, còn  request có trạng thái mặc định là pending rồi nè 

hoặc em dùng callback `before_create ` trong model request là đẹp, sao phải tốn công :v

Reply:
`Book.where(id: session[:book_ids])` cái where này chú viết scope trong model Book nhé `scope :by_ids, ->(ids){where id: ids) `

ngoài ra, nên có action filter là check `exits_book_ids_into_session?` nhé trước khi call tới action create
```
def exits_book_ids_into_session?
  # khi ko có sách nào mượn cả thì ko thể tới màn creatre dc 
  unless (session[:book_ids].any?)
     flash[:warning] = "Bạn chưa chọn sách nào cả"
    return redirect_to root_path
  end
  
   # check xem có id book nào ko exit trong DB ko
   @books = Book.find session[:book_ids]
rescue 
  flash[:error] = "Đã có sách không hợp lệ, vui lòng reload lại trang"
  redirect_to root_path
end
```

nếu làm ri, thì lúc load session[:book_ids] em phải kiểm tra có id nào ko hợp lệ trong session ko nhé 

nếu ko dùng find như trên thì em dùng where rồi check foreach kiểm tra coi có id nào invalid thì tự delete nó khỏi session[:book_ids] cũng dc